development-kit/pkg/engines/java/rules.go

```diff
@@ -16,13 +16,63 @@
 package java
 
 import (
+	engine "github.com/ZupIT/horusec-engine"
 	"github.com/ZupIT/horusec-engine/text"
-	"github.com/ZupIT/horusec/development-kit/pkg/enums/engine/advisories/java/and"
-	"github.com/ZupIT/horusec/development-kit/pkg/enums/engine/advisories/java/or"
-	"github.com/ZupIT/horusec/development-kit/pkg/enums/engine/advisories/java/regular"
+	"github.com/ZupIT/horusec/development-kit/pkg/engines/java/and"
+	"github.com/ZupIT/horusec/development-kit/pkg/engines/java/or"
+	"github.com/ZupIT/horusec/development-kit/pkg/engines/java/regular"
+	"github.com/ZupIT/horusec/development-kit/pkg/engines/jvm"
+	"github.com/ZupIT/horusec/development-kit/pkg/utils/logger"
 )
 
-func AllRulesJavaRegular() []text.TextRule {
+type Interface interface {
+	GetAllRules() (rules []engine.Rule)
+	GetTextUnitByRulesExt(projectPath string) ([]engine.Unit, error)
+}
+
+type Rules struct {
+	jvmRules jvm.Interface
+}
+
+func NewRules() Interface {
+	return &Rules{
+		jvmRules: jvm.NewRules(),
+	}
+}
+
+func (r *Rules) GetAllRules() (rules []engine.Rule) {
+	rules = r.addJavaRules(rules)
+	rules = r.jvmRules.GetAllRules(rules)
+	return rules
+}
+
+func (r *Rules) addJavaRules(rules []engine.Rule) []engine.Rule {
+	for index := range allRulesJavaAnd() {
+		rules = append(rules, allRulesJavaAnd()[index])
```

Comment:
I did a quick test to check the reading speed of this method, when using:
* Index: the method takes approximately 195.98ms to return its result
* Item: the method takes approximately 10.75ms to return its result
I think it would be interesting to use the approach of the item itself

Reply:
![image](https://user-images.githubusercontent.com/63816070/102882252-dbb11f00-442c-11eb-973a-90e434de5571.png)
![image](https://user-images.githubusercontent.com/63816070/102882312-f71c2a00-442c-11eb-9cad-9d7b6071ad42.png)

Reply:
Fixed